src/lib/db/feature-toggle-store.ts

```diff
@@ -171,34 +173,22 @@ export default class FeatureToggleStore implements IFeatureToggleStore {
 
     async setLastSeen(data: LastSeenInput[]): Promise<void> {
         const now = new Date();
-        const environmentArrays = this.mapMetricDataToEnvBuckets(data);
-        try {
-            for (const env of Object.keys(environmentArrays)) {
-                const toggleNames = environmentArrays[env].sort();
-                await this.db(FEATURE_ENVIRONMENTS_TABLE)
-                    .update({ last_seen_at: now })
-                    .where('environment', env)
-                    .whereIn(
-                        'feature_name',
-                        this.db(FEATURE_ENVIRONMENTS_TABLE)
-                            .select('feature_name')
-                            .whereIn('feature_name', toggleNames)
-                            .forUpdate()
-                            .skipLocked(),
-                    );
-
-                // Updating the toggle's last_seen_at also for backwards compatibility
-                await this.db(TABLE)
-                    .update({ last_seen_at: now })
-                    .whereIn(
-                        'name',
-                        this.db(TABLE)
-                            .select('name')
-                            .whereIn('name', toggleNames)
-                            .forUpdate()
-                            .skipLocked(),
-                    );
+        const enhancedData = data.map((value) => ({
+            feature_name: value.featureName,
+            last_seen_at: now,
+            environment: value.environment,
+        }));
+        const dataToPersist: typeof enhancedData = [];
+        for (const input of enhancedData) {
+            if (await this.exists(input.feature_name)) {
+                dataToPersist.push(input);
```

Comment:
This is not ideal.  I would much rather have this in the SQL query somehow

Reply:
we cannot do this. For customers with thousands of toggles it will overload the DB.

Reply:
because of the foreign key for feature_name, I think this can be dealt with in the conflict handling of the insert query itself.

Reply:
unfortunately not - we can either have ON CONFLICT DO NOTHING (which is needed for the constraint violation to `features` or ON CONFLICT (pkey) DO UPDATE (which is needed for upsert)